java/client/src/test/java/glide/api/RedisClusterClientTest.java

```diff
@@ -0,0 +1,101 @@
+package glide.api;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import glide.managers.CommandManager;
+import glide.managers.RedisExceptionCheckedFunction;
+import glide.managers.models.Command;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Test;
+import response.ResponseOuterClass.Response;
+
+public class RedisClusterClientTest {
+
+    @Test
+    @SneakyThrows
+    public void custom_command_returns_single_value() {
+        var commandManager = new TestCommandManager(null);
+
+        var client = new TestClient(commandManager, "TEST");
+
+        var value = client.customCommand(new String[0]).get();
+        assertAll(
+                () -> assertTrue(value.hasSingleData()),
+                () -> assertEquals("TEST", value.getSingleValue()));
+    }
+
+    @Test
+    @SneakyThrows
+    public void custom_command_returns_multi_value() {
+        var commandManager = new TestCommandManager(null);
+
+        var data = Map.of("key1", "value1", "key2", "value2");
+        var client = new TestClient(commandManager, data);
+
+        var value = client.customCommand(new String[0]).get();
+        assertAll(
+                () -> assertTrue(value.hasMultiData()), () -> assertEquals(data, value.getMultiValue()));
+    }
+
+    @Test
+    @SneakyThrows
+    // test checks that even a map returned as a single value when single node route is used
+    public void custom_command_with_single_node_route_returns_single_value() {
+        var commandManager = new TestCommandManager(null);
+
+        var data = Map.of("key1", "value1", "key2", "value2");
+        var client = new TestClient(commandManager, data);
+
+        var value = client.customCommand(new String[0], () -> true).get();
```

Comment:
I don't understand the Java syntax - why is there a function here, instead of a Route?
Is it working because this function implements the `Route` interface? If so, this isn't very accessible - I don't think that the alternative, of passing an actual route, is so complex as to justify this harm to readability.

Reply:
This is java tricks.
It is possible to define an anonymous class inline which extends existing one or implement an interface. When you're implementing an interface with a single function it could be even collapsed to lambda. Yeah, that could be confising.
I changed this in eae39763